packages/salesforcedx-vscode-core/test/vscode-integration/orgBrowser/metadataCmp.test.ts

```diff
@@ -248,4 +251,55 @@ describe('load metadata component data', () => {
     await cmpUtil.loadComponents(defaultOrg, metadataType, undefined, true);
     expect(cmdOutputStub.calledOnce).to.be.true;
   });
+
+  it('should validate that buildCustomObjectFieldsList() is called', async () => {
+    buildCustomObjectFieldsListStub.returns('');
+
+    const components = await cmpUtil.loadComponents(defaultOrg, 'CustomObject', 'DemoCustomObject', undefined);
+    expect(buildCustomObjectFieldsListStub.called).to.equal(true);
+    expect(buildCustomObjectFieldsListStub.calledWith('DemoCustomObject', defaultOrg, filePath)).to.be.true;
+  });
+
+  it('should validate that buildCustomObjectFieldsList() returns correctly formatted fields', async () => {
+    const fieldData = JSON.stringify({
+      result: {
+        fields: [
```

Comment:
Can we add an item for type: 'reference' ?

Reply:
done